docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Single Validator Stake Pool Manager Program
+---
+
+A work-in-progress program for permissionless management of stake pools that are
+only delegated to one validator.
+
+## Operation
+
+The single-validator manager program defines a canonical stake pool for any
+validator on the network, which can be created by anyone.
+
+The program manages the "delegation strategy" of the pool, which is to maximize
+the amount of stake on one validator, while maintaining a small amount of SOL
+in the reserve for small stakers.
+
+### Create Pool
+
+The `CreateStakePool` instruction allows anyone to create the canonical stake
+pool for a validator, at the cost of ~0.0124 SOL for account rent-exemption of
+the stake pool, validator list, reserve, mint, and fee account.
+
+After the pool is created, the reserve requires at least stake rent-exemption plus
+the minimum delegation amount to actually add the validator to the pool, which
+costs 1.0028288 SOL. The user receives pool tokens for this deposit.
+
+After the SOL deposit, the `AddValidatorToPool` instruction on the pool management
+program allows anyone to add the stake account for that validator.
+
+Once the added stake account is active, anyone may deposit stake accounts.
+
+### Depositing and withdrawing stake
+
+If stakers deposit or withdraw stake accounts from the pool, everything works
+seamlessly. Stakers receive pool tokens for their deposited stake account, and
+may use the pool tokens to withdraw a stake account.
```

Comment:
All stakes are merged into one on deposit, so there's only one stake account to withdraw from. When you withdraw a stake account, it goes into an uninitialized stake account that you provide.

You bring up a good point though, we could add the capability for the stake pool program to try a split *then* a merge if the destination account is already active, which would allow for tiny pool token amounts to be moved, rather than requiring a movement of `minimum_delegation_amount`. On the other hand, it would also require another "ephemeral" style account just for that instruction.

Reply:
seems like the pool instruction could be extended w/o breaking API by allowing the ephemeral account to be appended optionally?  that is, if transient already exists and it's not there, tx fails as it would today. but if we have ephemeral, try to split+merge?